lib/msf/core/exploit/remote/java/http/class_loader.rb

```diff
@@ -38,14 +41,16 @@ def on_request_uri(cli, request)
       return
     end
 
+    resource = request.raw_uri.delete_prefix(resource_uri)
```

Comment:
This line was causing some issues.

If `datastore['URIPATH']` is not set  when `resource_uri` calls `super`, [http_server.rb:648](https://github.com/rapid7/metasploit-framework/blob/ff0ecfa49ef8d4b7c0d2ada0e00d17aa31747afe/lib/msf/core/exploit/remote/http_server.rb#L648) returns a random string which doesn't exist in `request.raw_uri` as it's different from the one that gets generated by `start_service` here: [http_server.rb:149](https://github.com/rapid7/metasploit-framework/blob/ff0ecfa49ef8d4b7c0d2ada0e00d17aa31747afe/lib/msf/core/exploit/remote/http_server.rb#L149).  The prefix never gets deleted from `resource`, which causes the `case resource` statement below to default to the else block sending a 404. 

This quick fix got the module working for me: 
```suggestion
    resource = request.raw_uri.gsub(/^\/[^\/]+\//, ""))
```